tools/source_updater/updater/source.py

```diff
@@ -0,0 +1,164 @@
+import json
+import os
+import yaml
+from pydantic.error_wrappers import ValidationError
+from typing import Set
+
+from airbyte_cdk.connector import BaseConnector
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from updater.config import Config
+
+# This is needed as when interpreting SecretsManager, the path used is determined by the VERSION environment variable
+os.environ["VERSION"] = "dev"
+from ci_credentials import SecretsManager  # noqa: E402
+
+
+class SourceRepository:
+    def exists(self, source_name: str) -> bool:
+        return os.path.exists(self._path_from_name(source_name))
+
+    def fetch_catalog(self, source_name: str) -> ConfiguredAirbyteCatalog:
+        try:
+            return ConfiguredAirbyteCatalog.parse_obj(BaseConnector.read_config(self._catalog_path(source_name)))
+        except ValidationError as error:
+            raise ValueError(error)
+
+    def update_catalog(self, source_name: str, catalog: ConfiguredAirbyteCatalog) -> None:
+        with open(self._catalog_path(source_name), "w") as catalog_file:
+            json.dump(catalog.dict(exclude_none=True), catalog_file, indent=2, default=lambda x: x.value)
+
+    def delete_schemas_folder(self, source_name: str) -> None:
+        schemas_folder_path = os.path.join(self._path_from_name(source_name), source_name.replace("-", "_"), "schemas")
+        for root, dirs, files in os.walk(schemas_folder_path, topdown=False):
```

Comment:
nit: I think [`shutils.rmtree` with `ignore_errors=True` can do the traversal for you ](https://docs.python.org/3/library/shutil.html#shutil.rmtree)